harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -0,0 +1,374 @@
+from time import sleep
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from datetime import datetime, timedelta
+
+import pytest
+
+pytest_plugins = [
+    "harvester_e2e_tests.fixtures.api_client",
+    "harvester_e2e_tests.fixtures.images"
+]
+
+
+@pytest.fixture(scope="session")
+def virtctl(api_client):
+    code, ctx = api_client.vms.download_virtctl()
+
+    with NamedTemporaryFile("wb") as f:
+        f.write(ctx)
+        f.seek(0)
+        yield Path(f.name)
+
+
+@pytest.fixture(scope="session")
+def kubeconfig_file(api_client):
+    kubeconfig = api_client.generate_kubeconfig()
+    with NamedTemporaryFile("w") as f:
+        f.write(kubeconfig)
+        f.seek(0)
+        yield Path(f.name)
+
+
+@pytest.fixture(scope="module")
+def image_id(api_client, opensuse_image, unique_name, wait_timeout):
+    unique_image_id = f'image-{unique_name}'
+    code, data = api_client.images.create_by_url(
+        unique_image_id, opensuse_image.url, display_name=f"{unique_name}-{opensuse_image.name}"
+    )
+
+    assert 201 == code, (code, data)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.images.get(unique_image_id)
+        if 100 == data.get('status', {}).get('progress', 0):
+            break
+        sleep(3)
+    else:
+        raise AssertionError(
+            "Failed to create Image with error:\n"
+            f"Status({code}): {data}"
+        )
+
+    yield f"{data['metadata']['namespace']}/{unique_image_id}"
+
+    code, data = api_client.images.delete(unique_image_id)
+
+
+@pytest.fixture(scope="module")
+def unique_vm_name(unique_name):
+    return f"vm-{unique_name}"
+
+
+@pytest.mark.p0
+@pytest.mark.virtualmachines
+@pytest.mark.dependency(name="minimal_vm")
+def test_minimal_vm(api_client, image_id, unique_vm_name, wait_timeout):
+    cpu, mem = 1, 2
+    vm = api_client.vms.Spec(cpu, mem)
+    vm.add_image("disk-0", image_id)
+
+    code, data = api_client.vms.create(unique_vm_name, vm)
+
+    assert 201 == code, (code, data)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(unique_vm_name)
+        if 200 == code and "Running" == data.get('status', {}).get('phase'):
+            break
+        sleep(3)
+    else:
+        raise AssertionError(
+            f"Failed to create Minimal VM({cpu} core, {mem} RAM) with errors:\n"
+            f"Phase: {data.get('status', {}).get('phase')}\t"
+            f"Status: {data.get('status')}\n"
+            f"API Status({code}): {data}"
+        )
+
+
```

Comment:
There needs to be a test skel here with the steps per [test skel instructions](https://github.com/harvester/tests#user-content-backend-test-skeletons). I took a look at the subsequent tests and see why there aren't test skels for all of them (they are small and self expalantory). However we should probably add a more overarching test skel with steps at the higher level then.